src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py

```diff
@@ -374,11 +374,12 @@ def remove(self, retainsnaps=False, internal_cleanup=False):
                 self.auth_mdata_mgr.delete_subvolume_metadata_file(self.group.groupname, self.subvolname)
                 return
         if self.state != SubvolumeStates.STATE_RETAINED:
-            self.trash_incarnation_dir()
+            subvolume_path = self.path
             self.metadata_mgr.remove_section(MetadataManager.USER_METADATA_SECTION)
```

Comment:
Should we move this below? In case of osd full, if removing user metadata section succeeds and updating a section fails, it is still inconsistent. Could you please test this out ?

Reply:
Also the caller seems to be catching only VolumeException. In this case, if the osd is full, the configparser related exceptions needs to be caught ? Please test it out.

Reply:
@neesingh-rh ping?

Reply:
@neesingh-rh I think this is still not taken care ? Am I missing something ? Did you validate this ?

Reply:
> Also the caller seems to be catching only VolumeException. In this case, if the osd is full, the configparser related exceptions needs to be caught ? Please test it out.

@kotreshhr I didn't really get this?

Reply:
PTAL